tests: posix: common: fix test for cross-clock equality

pabigot opened the pull request:
Distinct reads of the clock are not guaranteed to be equal to the resolution of the clock.  Since the clock source is the a millsecond-resolution uptime clock just do this in a loop until a case is found where the system clock didn't advance during the POSIX clock reads.

Fixes #25453

pabigot committed:
tests: posix: common: fix test for cross-clock equality

Distinct reads of the clock are not guaranteed to be equal to the
resolution of the clock.  Since the clock source is the a
millsecond-resolution uptime clock just do this in a loop until a case
is found where the system clock didn't advance during the POSIX clock
reads.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

ABOSTM reviewed (commented):
LGTM
As I understand it replaces #25454, right ?
Is there no risk to have an endless loop ?

pabigot commented:
> LGTM
> As I understand it replaces #25454, right ?

Huh; sorry, no, I missed that one.  I'd thought there was a fix but got it mixed up with nanosleep.

We can prefer #25454 if somebody will override the NACK that's been addressed.  Marking this DNM until we get that resolved.

> Is there no risk to have an endless loop ?

Not in any system where tests will pass, IMO.  And if it it does, well, that's a failure too.

carlescufi committed:
Revert "github: labeler: Fix dts/ folder matching"

This reverts commit d20b6f90eab85110233a39d98b296a72ad2a28c5.

carlescufi committed:
github: labeler: Remove line using negation (!)

The current version of the labeler action doesn't support yet negations.
This is supported in master but we probably want to wait until a release
to update. In the meantime remove the rule that uses negations.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

pabigot commented:
Closing; #28516 has more velocity.
